src/compiler/checker.ts

```diff
@@ -27253,7 +27257,8 @@ namespace ts {
         }
 
         function checkQualifiedName(node: QualifiedName, checkMode: CheckMode | undefined) {
-            return checkPropertyAccessExpressionOrQualifiedName(node, node.left, checkNonNullExpression(node.left), node.right, checkMode);
+            const leftType = isTypeQueryNode(node.parent) && isThisIdentifier(node.left) ? checkNonNullType(checkThisExpression(node.left), node.left) : checkNonNullExpression(node.left);

```

Comment:
does this work when the QualifiedName is nested, like `this.x.y`? I think you need a test for this.

Reply:
It doesn't work for `this.x.y`. I've replaced `isTypeQueryNode` with `isPartOfTypeQuery`